tools/site_admin/copy_pages.php

```diff
@@ -231,17 +231,21 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
         die( "Error: no projectid data supplied" );
     }
 
+    $project_obj = [
+        "from" => new Project($projectid_["from"]),
+        "to" => new Project($projectid_["to"]),
+    ];
+
     foreach ( array( 'from', 'to' ) as $which )
     {
-        $projectid = $projectid_[$which];
-        $project = new Project($projectid);
+        $project = $project_obj[$which];
 
         if(!$project->is_utf8)
         {
-            die("Project table $projectid is not UTF-8.");
+            die("Project table {$project->projectid} is not UTF-8.");
```

Comment:
Does the project constructor not do this for us?

Reply:
No, it can't. We did not convert project tables to UTF-8 for posted or archived projects. If those projects get un-posted or un-archived (which is rare but does happen) they will need to be converted to UTF-8 before being used. We have a few places in the code that check for UTF-8-ness and a SA script that allows SAs to convert them.